### PR TITLE
Add new `Performance/ReverseFirst` cop

### PR DESCRIPTION
**Pull request by fatkodima**
There is no need to reverse the whole array to get last n items in order from end to begin.

```ruby
# bad
array.reverse.first(5)
array.reverse.first

# good
array.last(5).reverse
array.last
```

### Benchmark

```ruby
require 'bundler/inline'

gemfile(true) do
  gem 'benchmark-ips'
end

ARRAY = (1..100).to_a

Benchmark.ips do |x|
  x.report('Array.last(n).reverse')  { ARRAY.last(5).reverse }
  x.report('Array.reverse.first(n)') { ARRAY.reverse.first(5) }
  x.compare!
end
```

### Results

```
Warming up --------------------------------------
Array.last(n).reverse
                       515.468k i/100ms
Array.reverse.first(n)
                       101.975k i/100ms
Calculating -------------------------------------
Array.last(n).reverse
                          4.544M (± 8.7%) i/s -     22.681M in   5.034133s
Array.reverse.first(n)
                        913.988k (± 9.9%) i/s -      4.589M in   5.080680s

Comparison:
Array.last(n).reverse:  4543639.0 i/s
Array.reverse.first(n):   913988.0 i/s - 4.97x  (± 0.00) slower
```

Obviously, this scales linearly with array size.

### CODE REVIEW

File: spec/rubocop/cop/performance/reverse_first_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Performance::ReverseFirst do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense and corrects when using `#reverse.first(5)`' do
+    expect_offense(<<~RUBY)
+      array.reverse.first(5)
+            ^^^^^^^^^^^^^^^^ Use `last(5).reverse` instead of `reverse.first(5)`.
+    RUBY
```
Comment (koic) [inline spec/rubocop/cop/performance/reverse_first_spec.rb:10]: Can you add a test for auto-correction?

File: spec/rubocop/cop/performance/reverse_first_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it 'registers an offense and corrects when using `#reverse.first`' do
+    expect_offense(<<~RUBY)
+      array.reverse.first
+            ^^^^^^^^^^^^^ Use `last` instead of `reverse.first`.
+    RUBY
```
Comment (koic) [inline spec/rubocop/cop/performance/reverse_first_spec.rb:17]: Ditto.

**Comment by fatkodima** [comment]
Updated.

**Comment by koic** [comment]
Thanks!

**Pull request merged**